pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```
@@ -15,30 +14,53 @@ self: super:
 let
   inherit (neovimUtils) grammarToPlugin;
 
-  overrides = prev: {
+  buildQueries =
+    { language }:
+    runCommand "nvim-treesitter-queries-${language}"
+      {
+        passthru = { inherit language; };
+        meta.description = "Queries for ${language} from nvim-treesitter";
+      }
+      ''
+        mkdir -p "$out/queries"
+        if [ -d "${super.nvim-treesitter.src}/runtime/queries/${language}" ]; then
+          ln -s "${super.nvim-treesitter.src}/runtime/queries/${language}" "$out/queries/${language}"
+        else
+          echo "Error: there is no queries for ${language}."
```

quibble: 

```suggestion
          echo "Error: there are no queries for ${language}."
```

issue: I'm still not sure if we want to fail if a parser exists without queries. There may be a use case for that and nvim-treesitter [seems to check if the queries exist before trying to copy them](https://github.com/nvim-treesitter/nvim-treesitter/blob/4fc09bee78e91bf4ba471cdab4bf9dfa37fde51c/lua/nvim-treesitter/install.lua#L422). So in principle, it's possible for nvim-treesitter to ship parsers without queries.